interfaces/builtin/network_status.go

```diff
@@ -24,131 +24,82 @@ import (
 
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/apparmor"
-	"github.com/snapcore/snapd/interfaces/dbus"
 	"github.com/snapcore/snapd/snap"
 )
 
-const networkStatusSummary = `allows operating as the NetworkingStatus service`
+const networkStatusSummary = `allows access to network connectivity status`
 
 const networkStatusBaseDeclarationSlots = `
   network-status:
     allow-installation:
       slot-snap-type:
         - app
+        - core
     deny-connection: true
 `
 
 const networkStatusPermanentSlotAppArmor = `
-# Description: Allow owning the NetworkingStatus bus name on the system bus
-
-# DBus accesses
-#include <abstractions/dbus-strict>
-
-dbus (send)
-   bus=system
-   path=/org/freedesktop/DBus
-   interface=org.freedesktop.DBus
-   member={Request,Release}Name
-   peer=(name=org.freedesktop.DBus, label=unconfined),
-
-dbus (bind)
-   bus=system
-   name="com.ubuntu.connectivity1.NetworkingStatus",
-
-# allow queries from unconfined
-dbus (receive)
-    bus=system
-    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
-    interface=org.freedesktop.DBus.*
-    peer=(label=unconfined),
+# Description: allow providing network connectivity status
 `
 
 const networkStatusConnectedSlotAppArmor = `
-# Description: allow access to NetworkingStatus service
-
-dbus (receive)
-    bus=system
-    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
-    interface=org.freedesktop.DBus.*
-    peer=(label=###PLUG_SECURITY_TAGS###),
+# Description: allow providing network connectivity status
 `
 
 const networkStatusConnectedPlugAppArmor = `
-# Description: Allow using NetworkingStatus service.
-
-#include <abstractions/dbus-strict>
+# Description: allow access to network connectivity status
 
-# Allow all access to NetworkingStatus service
-dbus (send)
-    bus=system
-    interface=com.ubuntu.connectivity1.NetworkingStatus{,*}
-    path=/com/ubuntu/connectivity1/NetworkingStatus
-    peer=(label=###SLOT_SECURITY_TAGS###),
+#include <abstractions/dbus-session-strict>
```

Comment:
To not break existing snaps, we need to add:
```
# While not needed for org.freedesktop.portal.NetworkMonitor, snapd
# < 2.45 supported connectivity-api on the system bus and existing
# plugging snaps would get this rule so preserve it to avoid
# regressions. While unneeded, the rule doesn't appreciably expand
# the access (esp. when considering other auto-connected interfaces).
#include <abstractions/dbus-strict>
```

Reply:
Is this necessary, given that existing users could only have taken advantage of the old connected plug AppArmor rules if they actually managed to connect their network-status plug?

Reply:
Actually, you're right. It is not, network-status is not connectable.